apps/worker/src/app/workflow/usecases/send-message/send-message.base.ts

```diff
@@ -54,12 +55,10 @@ export abstract class SendMessageBase extends SendMessageType {
         ...command,
       }),
   })
-  protected async getIntegration(getDecryptedIntegrationsCommand: GetDecryptedIntegrationsCommand) {
-    return (
-      await this.getDecryptedIntegrationsUsecase.execute(
-        GetDecryptedIntegrationsCommand.create(getDecryptedIntegrationsCommand)
-      )
-    )[0];
+  protected async getIntegration(
+    selectIntegrationCommand: SelectIntegrationCommand
```

Comment:
In the child use cases you are not creating the commands, just passing an object. Even if they are compatible what are the reasons to do it like that?

Reply:
Do you mean to call the parent method in the children's classes?

If so then this is basically Legacy code where we encapsulated the functionality with @Cached decorator. I agree that this one looks like overkill at the moment but once we add the Feature Flag it will make more sense IMO.

Reply:
No, that we are not using <X>Command to create the objects in the use cases to pass the information to the base use case.

Reply:
In the children's classes we are not using the use-cases, that's the reason. We do call this parent function `getIntegration` and it calls the use-case and uses the object and creates the command.